src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -150,7 +156,11 @@ def _load_profile_stats(cls, area_dict: Dict, subdict: Dict, core_stats={},
 
         cls._calc_min_max_from_sim_dict(subdict, key_name)
 
-    def update(self, area_result_dict={}, core_stats={}, current_market_slot=None):
+    def update(self, area_result_dict=None, core_stats=None, current_market_slot=None):
+        if area_result_dict is None:
+            area_result_dict = {}
+        if core_stats is None:
+            core_stats = {}
```

Comment:
Can you tell me the need for this change?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments